cypress/integration/search/post_search_display.js

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+/*eslint max-nested-callbacks: ["error", 3]*/
+
+describe('Post search display', () => {
+    it('S14252 After clearing search query, search options display', () => {
+        // # Login and navigate to the app
+        cy.apiLogin('user-1');
+        cy.visit('/');
+        const searchWord = 'Hello';
+
+        // # post message
+        cy.postMessage('Hello');
+
+        // # search word in searchbox
+        cy.get('#searchBox').type(searchWord).type('{enter}');
+
+        // # searchbox should have attribute 'value'
+        cy.get('#searchBox').should('have.attr', 'value', searchWord);
+
+        // # click on "x" displayed on searchbox
+        cy.get('#searchClearButton').click();
+
+        // # RHS should be visible with search results
+        cy.get('.sidebar-right__body').should('be.visible');
+
+        // # click on searchbox
+        cy.get('#searchBox').click();
+
+        // # focused element should be searchbox
+        cy.focused().should('have.attr', 'id', 'searchBox');
+
+        // # search options menu is visible
+        cy.get('#searchbar-help-popup').should('have.class', 'visible');
+
+        // # check the contents in search options
+        cy.get('#searchbar-help-popup').contains('Search Options');
```

Comment:
To piggy back on the above conversation, we should be able to easily validate the search options, let me know if you need help with that. The `id` of the search help box should be `searchbar-help-popup`. Also, `contains` should be used for getting elements, and generally when we only have text to base our element selection on.

Reply:
how we can validate the search options ? would you please help on this ?

Reply:
Not sure what @thekiiingbob is suggesting but one way is something like:
```javascript
cy.get('#searchbar-help-popup').within(() => {
    cy.get('h4 span').first().should('have.text', 'Search Options');
    cy.get('span ul li').first().should('have.text', 'Use "quotation marks" to search for phrases');
    cy.get('span ul li').eq(1).should('have.text', 'Use from: to find posts from specific users and in: to find posts in specific channels');
    cy.get('span ul li').eq(2).should('have.text', 'Use on: to find posts on a specific date');
    cy.get('span ul li').eq(3).should('have.text', 'Use before: to find posts before a specific date');
    cy.get('span ul li').last().should('have.text', 'Use after: to find posts after a specific date');
});
```